tests/example_data/data_loading/csv_dataset_loader.py

```diff
@@ -0,0 +1,110 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+from __future__ import annotations
+
+import os.path
+from pathlib import Path
+from typing import List, TYPE_CHECKING
+from urllib.parse import urlparse
+
+import pandas as pd
+
+from superset import config, db
+from superset.utils.database import get_example_database
+
+if TYPE_CHECKING:
+    from superset.connectors.sqla.models import SqlaTable
+
+
+class CsvDatasetLoader:
+    # A simple csvloader, should run in Superset AppContext
+    csv_path: str
+    df: pd.DataFrame
+    table_name: str
+    dataset: SqlaTable
+
+    def __init__(
+        self,
+        csv_path: str,
+        cache: bool = True,
+        parse_dates: List[str] = [],
+    ):
+        # read from http
+        if csv_path.startswith("http") and csv_path.endswith(".csv"):
+            filename = urlparse(csv_path).path.split("/")[-1]
+            filepath = os.path.join(config.DATA_DIR, filename)
+            if os.path.exists(filepath) and cache:
+                self.csv_path = filepath
+                self.df = pd.read_csv(filepath, parse_dates=parse_dates)
+                self.table_name = filename.replace(".csv", "")
+            else:
+                self.df = pd.read_csv(csv_path, parse_dates=parse_dates)
+                if cache:
+                    self.df.to_csv(filepath, index=False)
+                self.csv_path = filepath
+                self.table_name = filename.replace(".csv", "")
+
+        # read from fs
+        if os.path.exists(csv_path) and csv_path.endswith(".csv"):
+            self.csv_path = csv_path
+            self.df = pd.read_csv(csv_path, parse_dates=parse_dates)
+            self.table_name = Path(csv_path).name.replace(".csv", "")
+
+    def load_table(self) -> None:
+        # load table to the default schema
+        example_database = get_example_database()
+        self.df.to_sql(
+            name=self.table_name,
+            con=example_database.get_sqla_engine(),
+            index=False,
+            if_exists="replace",
+        )
```

Comment:
We can probably reuse the logic from https://github.com/apache/superset/blob/a2d34ec4b8a89723e7468f194a98386699af0bd7/superset/datasets/commands/importers/v1/utils.py#L151-L182 here (you just need to modify it to load local files).

Reply:
Nice! I will do it.

Reply:
These 2 functions look different after I dig into them. We don't have a **Dataset** before we call `CsvDatasetLoader`. The `CsvDatasetLoader` loads a CSV to **example database**, and then infer to **Dataset** from the loaded table. The `load_data` has existed **Dataset**.

Reply:
You're right, but it might not be hard to modify it so it doesn't need the dataset (it's using it just to infer types). But maybe later we can consolidate these two functions into one (and maybe also `superset.examples.helpers.get_example_data`).

Reply:
Okay. I will try to extend `load_data`, and reuse it.